distribution/dashboard/internal/handler/metric_handler.go

```diff
@@ -22,23 +24,21 @@ import (
 // @Failure 500 object response.APIResponse
 // @Router /api/v1/metrics [GET]
 // @Security ApiKeyAuth
-func ListMetricMetas(c *gin.Context) {
+func ListMetricMetas(c *gin.Context) ([]response.MetricClass, error) {
 	// return mock data
 	language := c.GetHeader("Accept-Language")
 	scope := c.Query("scope")
 	if scope != metricconst.ScopeCluster && scope != metricconst.ScopeTenant && scope != metricconst.ScopeClusterOverview {
 		err := errors.New("invalid scope")
 		logHandlerError(c, err)
-		SendBadRequestResponse(c, nil, err)
-	} else {
-		metricClasses, err := metric.ListMetricClasses(scope, language)
-		if err != nil {
-			logHandlerError(c, err)
-			SendInternalServerErrorResponse(c, nil, err)
-		} else {
-			SendSuccessfulResponse(c, metricClasses)
-		}
+		return nil, oberr.NewBadRequest(err.Error())
 	}
+	metricClasses, err := metric.ListMetricClasses(scope, language)
+	if err != nil {
+		logHandlerError(c, err)
```

Comment:
logHandlerError can be moved into wrapper, thus we can log all the errors in one place, all the handlers only need to return the error

Reply:
Good idea! I have removed all loggings in specific handlers and moved error logging logic into the handler wrapper.